client/matching/client.go

```diff
@@ -232,6 +232,33 @@ func (c *clientImpl) ListTaskListPartitions(
 	return client.ListTaskListPartitions(ctx, request, opts...)
 }
 
+func (c *clientImpl) GetTaskListsForDomain(
+	ctx context.Context,
+	request *types.MatchingGetTaskListsForDomainRequest,
+	opts ...yarpc.CallOption,
+) (*types.GetTaskListsForDomainResponse, error) {
+	opts = common.AggregateYarpcOptions(ctx, opts...)
+	clients := c.clients.GetAllTaskLists()
+
+	var response *types.GetTaskListsForDomainResponse
+	for _, tl := range clients {
```

Comment:
call getTaskListForDomain for each client?

Reply:
Yes. I am calling getTaskListForDomain for each client.

Reply:
After getting all clients from GetAllClients, you don't need to call getClientForTaskList

Reply:
Updated.